src/Akka.Persistence.Hosting/AkkaPersistenceHostingExtensions.cs

```diff
@@ -37,13 +38,29 @@ public sealed class AkkaPersistenceJournalBuilder
         internal readonly AkkaConfigurationBuilder Builder;
         internal readonly Dictionary<Type, HashSet<string>> Bindings = new Dictionary<Type, HashSet<string>>();
         internal readonly Dictionary<string, Type> Adapters = new Dictionary<string, Type>();
+        internal AkkaHealthCheckRegistration? HealthCheckRegistration = null;
 
         public AkkaPersistenceJournalBuilder(string journalId, AkkaConfigurationBuilder builder)
         {
             JournalId = journalId;
             Builder = builder;
         }
 
+        /// <summary>
+        /// Uses the built-in journal health check on the Akka.Persistence.Journal.
+        /// </summary>
+        /// <param name="unHealthyStatus">Default status to return when the plugin reports <see cref="PersistenceHealthStatus.Unhealthy"/>
+        /// or <see cref="PersistenceHealthStatus.Degraded"/>. Defaults to degraded.</param>
+        /// <param name="name">Optional name to add to the health check.</param>
+        /// <returns></returns>
+        public AkkaPersistenceJournalBuilder WithHealthCheck(HealthStatus unHealthyStatus = HealthStatus.Degraded,
```

Comment:
Registers the Akka.Persistence health check definition is inside the `AsyncWriteJournal` - by default it just looks as the state of the `CircuitBreaker`, which is a "good enough" heuristic typically.